### ISSUE TITLE
Mixins have inconsistent order

### ISSUE THREAD

**Issue by zapov** [castleproject/Core#474]
In MixinData interfaces are sorted as 

    sortedMixedInterfaceTypes.Sort((x, y) => x.FullName.CompareTo(y.FullName));

instead of

    sortedMixedInterfaceTypes.Sort((x, y) => String.CompareOrdinal(x.FullName, y.FullName));

or even better via

    sortedMixedInterfaceTypes.Sort(new FieldReferenceComparer());

where `FieldReferenceComparer` could be renamed to `TypeNameComparer`

This causes wrong ctor order and fails to create an instance with strange message due to bad internal order ;(

I fixed my immediate problem by changing names, but it would be nice if this is fixed eventually.

**Comment by stakx** [comment]
> This causes wrong ctor order and fails to create an instance with strange message due to bad internal order ;(

Could you please post a short code example that demonstrates this?

**Comment by jonorossi** [comment]
Thanks @zapov.

This is the output of the new unit test in #475 without the fix:

```
Test Name:	CanCreateMixinWithCaseSensitiveImplementationOrder
Test FullName:	Castle.DynamicProxy.Tests.MixinTestCase.CanCreateMixinWithCaseSensitiveImplementationOrder
Test Source:	C:\src\castle-core\src\Castle.Core.Tests\DynamicProxy.Tests\MixinTestCase.cs : line 405
Test Outcome:	Failed
Test Duration:	0:00:04.976

Result StackTrace:	
  at Castle.DynamicProxy.ProxyGenerator.CreateClassProxyInstance(Type proxyType, List`1 proxyArguments, Type classToProxy, Object[] constructorArguments)
   at Castle.DynamicProxy.ProxyGenerator.CreateClassProxy(Type classToProxy, Type[] additionalInterfacesToProxy, ProxyGenerationOptions options, Object[] constructorArguments, IInterceptor[] interceptors)
   at Castle.DynamicProxy.ProxyGenerator.CreateClassProxy(Type classToProxy, ProxyGenerationOptions options, Object[] constructorArguments, IInterceptor[] interceptors)
   at Castle.DynamicProxy.Tests.MixinTestCase.CanCreateMixinWithCaseSensitiveImplementationOrder()

Result Message:	
Castle.DynamicProxy.InvalidProxyConstructorArgumentsException : Can not instantiate proxy of class: Castle.DynamicProxy.Tests.Mixins.MixinDomainObject.
Could not find a constructor that would match given arguments:
System.String
```

**Comment by jonorossi** [comment]
Fixed by #475.
